operator/cmd/root.go

```diff
@@ -106,9 +107,27 @@ var (
 			}
 		}),
 
+		cell.Provide(func(
+			operatorCfg *operatorOption.OperatorConfig,
+		) api.SharedConfig {
+			return api.SharedConfig{
+				EnableK8s: operatorCfg.EnableK8s,
```

Comment:
I think it'd be preferable to this `SharedConfig` to instead check for k8s enabled status via the k8s clientset: `clientset.IsEnabled()`, rational being that it's better to have only a single source of truth. 

Hmm, or is the problem that the k8s client gets configured and we want to use `EnableK8s` to overriding? Alternative I suppose is to explicitly `Disable()` the client based on `EnableK8s` in an `Invoke`. That would also make sure that the heartbeats aren't running. Though need to be careful that the `Invoke` runs early enough before any calls to `clientset.IsEnabled()`.

... almost feels like the cleanest solution would be to make `enable-k8s` part of `client.Config`, but that would also then introduce it to `cilium-agent` and others. Still maybe not a bad idea?